backend/dto/posts.js

```diff
@@ -0,0 +1,22 @@
+import models from '../database/models'
+
+/* eslint-disable import/prefer-default-export */
+export const convertToPostDto = async (post) => {
+    const author = await models.users.findByPk(post.author)
+    const children = await models.posts.findAll({
+        attributes: ['id'],
+        where: { parent: post.id },
+    })
+
+    return {
+        id: post.id,
+        content: post.text_content,
+        author,
+        parent: post.parent,
+        children: children.map((child) => child.id),
+        usersLiked: 0, // work in progress.
+        usersShared: 0, // work in progress.
```

Comment:
I recommend finishing that part first before merging the PR

Reply:
Fixed it. Now it shows the actual number of likes and shares. Had to fix the model as well.